### PR TITLE
chore: replace @testing-library/react-hooks by @testing-library/react

### PR DESCRIPTION
**Pull request by gre**
### 📝 Description

in our test you see a lot of

      Warning: ReactDOM.render is no longer supported in React 18. Use createRoot instead. Until you switch to the new API, your app will behave as if it's running React 17. Learn more: https://reactjs.org/link/switch-to-createroot
to fix it, we simply replace @testing-library/react-hooks by @testing-library/react 

see [GitHub - testing-library/react-hooks-testing-library: 🐏 Simple and complete React hooks testing utilities that encourage good testing practices.](https://github.com/testing-library/react-hooks-testing-library?tab=readme-ov-file#a-note-about-react-18-support)

### ❓ Context

- **JIRA or GitHub link**: https://ledgerhq.atlassian.net/browse/LIVE-10961

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [ ] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - tests

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

**Review by Justkant — DISMISSED** [review]
Nice thx for the update

**Comment by gre** [comment]
TiL live-common have hybrid js environement in our Jest config. As it's too implicit and magical, I chose to explicit add a comment 

```
/**
 * @jest-environment jsdom
 */
 ```
 
 in future, we need to move them to dedicated libraries (split the react parts out) so we will be able to reconfigure the react libs to be on jsdom. in the meantime it's better to try to minimize as much as possible the specifics from our jest config...

**Review by mcayuelas-ledger — DISMISSED** [review]

File: libs/ledger-live-common/src/families/evm/react.react.test.ts [hunk 1]
```diff
@@ -1,5 +1,8 @@
+/**
```
Comment (KVNLS) [inline libs/ledger-live-common/src/families/evm/react.react.test.ts:1]: I think that you can rename this file `react.test.js`
react.test.js pattern was used to apply @jest-environment jsdom

File: libs/ledger-live-common/src/deviceSDK/hooks/useToggleOnboardingEarlyChecks.test.ts [hunk 2]
```diff
@@ -1,4 +1,7 @@
-import { renderHook, act } from "@testing-library/react-hooks";
+/**
+ * @jest-environment jsdom
+ */
```
Comment (KVNLS) [inline libs/ledger-live-common/src/deviceSDK/hooks/useToggleOnboardingEarlyChecks.test.ts:3]: Thanks for managing the env like this

File: libs/ledger-live-common/src/families/cosmos/datasets/__snapshots__/coreum.integration.test.ts.snap [hunk 3]
```diff
@@ -85,108 +85,8 @@ exports[`coreum currency bridge scanAccounts coreum seed 1 1`] = `
...
-      ],
-      "standard": undefined,
-      "tokenId": undefined,
-      "transactionSequenceNumber": 0,
-      "type": "IN",
-      "value": "1000000",
-    },
-  ],
+  [],
+  [],
```
Comment (Justkant) [inline libs/ledger-live-common/src/families/cosmos/datasets/__snapshots__/coreum.integration.test.ts.snap:89]: Is this change expected ?

**Review by Justkant — COMMENTED** [review]
LGTM just wondering about this snapshot change

**Review by Justkant — DISMISSED** [review]

**Review by Justkant — APPROVED** [review]

**Review by valpinkman — APPROVED** [review]

**Pull request merged**